dev/run-tests-jenkins.py

```diff
@@ -0,0 +1,253 @@
+
+    if http_code and not http_code == 201:
+        print_err(" > http_code:", http_code)
+        print_err(" > api_response:", api_response)
+        print_err(" > data:", posted_message)
+
+    if curl_returncode == 0 and http_code == 201:
+        print(" > Post successful.")
+
+
+def send_archived_logs():
+    print("Archiving unit tests logs...")
+
+    log_files = run_cmd(['find', '.',
+                         '-name', 'unit-tests.log',
+                         '-o', '-path', './sql/hive/target/HiveCompatibilitySuite.failed',
+                         '-o', '-path', './sql/hive/target/HiveCompatibilitySuite.hiveFailed',
+                         '-o', '-path', './sql/hive/target/HiveCompatibilitySuite.wrong'],
+                        return_output=True)
+
+    if log_files:
+        log_archive = "unit-tests-logs.tar.gz"
+
+        run_cmd(['tar', 'czf', log-archive] + log_files.strip().split('\n'))
+
+        jenkins_build_dir = os.environ["JENKINS_HOME"] + "/jobs/" + os.environ["JOB_NAME"] + \
+            "/builds/" + os.environ["BUILD_NUMBER"]
+
+        scp_proc = subprocess.Popen(['scp', log_archive,
+                                     'amp-jenkins-master:' + jenkins_build_dir + '/' + log_archive],
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+        scp_stdout, scp_stderr = scp_proc.communicate()
+        scp_returncode = scp_proc.returncode
+
+        if not scp_returncode == 0:
+            print_err("Failed to send archived unit tests logs to Jenkins master.")
+            print_err(" > scp_status:",  scp_returncode)
+            print_err(" > scp_output:", scp_stdout)
+        else:
+            print(" > Send successful.")
+    else:
+        print_err(" > No log files found.")
+
+        rm_r(log_archive)
+
+
+def run_pr_tests(pr_tests, ghprb_actual_commit, sha1):
+    # Ensure we save off the current HEAD to revert to
+    current_pr_head = run_cmd(['git', 'rev-parse', 'HEAD'], return_output=True).strip()
+    pr_results = list()
+
+    for pr_test in pr_tests:
+        pr_results.append(run_cmd(['bash', os.path.join(SPARK_HOME, 'dev', 'tests', pr_test),
+                                   ghprb_actual_commit, sha1],
+                                  return_output=True).strip())
+        # Ensure, after each test, that we're back on the current PR
+        run_cmd(['git', 'checkout', '-f', current_pr_head])
+    return pr_results
+
+
+def bind_message_base(build_display_name, build_url, ghprb_pull_id, short_commit_hash, commit_url):
+    """
+    Given base parameters to generate a strong Github message response, binds those
+    parameters into a closure without the specific message and returns a function
+    able to generate strong messages for a specific description.
+    """
+    return lambda mssg, post_mssg="": \
+        '**[Test build ' + build_display_name + ' ' + mssg + '](' + build_url + \
```

Comment:
This code is really hard to read.  Can you use multi-line string formatting and interpolation to make this easier to read?